Userland/Libraries/LibWeb/HTML/HTMLCanvasElement.cpp

```diff
@@ -35,40 +35,100 @@ unsigned HTMLCanvasElement::height() const
     return attribute(HTML::AttributeNames::height).to_uint().value_or(150);
 }
 
+void HTMLCanvasElement::reset_context_to_default_state()
+{
+    m_context.visit(
+        [](NonnullRefPtr<CanvasRenderingContext2D>& two_dimensional_context) {
+            two_dimensional_context->reset_to_default_state();
+        },
+        [](NonnullRefPtr<WebGL::WebGLRenderingContext>&) {
+            TODO();
+        },
+        [](Empty) {
+            // Do nothing.
+        });
+}
+
 void HTMLCanvasElement::set_width(unsigned value)
 {
     set_attribute(HTML::AttributeNames::width, String::number(value));
     m_bitmap = nullptr;
-    if (m_context)
-        m_context->reset_to_default_state();
+    reset_context_to_default_state();
 }
 
 void HTMLCanvasElement::set_height(unsigned value)
 {
     set_attribute(HTML::AttributeNames::height, String::number(value));
     m_bitmap = nullptr;
-    if (m_context)
-        m_context->reset_to_default_state();
+    reset_context_to_default_state();
 }
 
 RefPtr<Layout::Node> HTMLCanvasElement::create_layout_node(NonnullRefPtr<CSS::StyleProperties> style)
 {
     return adopt_ref(*new Layout::CanvasBox(document(), *this, move(style)));
 }
 
-CanvasRenderingContext2D* HTMLCanvasElement::get_context(String type)
+HTMLCanvasElement::HasOrCreatedContext HTMLCanvasElement::try_create_2d_context()
+{
+    if (!m_context.has<Empty>())
+        return m_context.has<NonnullRefPtr<CanvasRenderingContext2D>>() ? HasOrCreatedContext::Yes : HasOrCreatedContext::No;
+
+    m_context = CanvasRenderingContext2D::create(*this);
+    return HasOrCreatedContext::Yes;
+}
+
+JS::ThrowCompletionOr<HTMLCanvasElement::HasOrCreatedContext> HTMLCanvasElement::try_create_webgl_context(JS::Value options)
```

Comment:
Same here, even though this can fail, `TRY(try_...)` is strange and usually only used when we have fallible and infallible variants during a transition period :)